include/zephyr/bluetooth/classic/map_client.h

```diff
@@ -0,0 +1,173 @@
+	MAP_PARAMETER_MASK_SEND = (0x1 << 13),
+	MAP_PARAMETER_MASK_PROTECTED = (0x1 << 14),
+	MAP_PARAMETER_MASK_REPLYTO_ADDRESSING = (0x1 << 15),
+};
+
+enum {
+	MAP_MESSAGE_RESULT_TYPE_DATETIME = 0x01,
+	MAP_MESSAGE_RESULT_TYPE_BODY = 0x02,
+};
+
+struct bt_map_client;
+
+struct bt_map_result {
+	uint8_t type;
+	uint16_t len;
+	uint8_t *data;
+};
+
+struct bt_map_client_cb {
+	void (*connected)(struct bt_map_client *client);
+	void (*disconnected)(struct bt_map_client *client);
+	void (*set_path_finished)(struct bt_map_client *client);
+	void (*recv)(struct bt_map_client *client, struct bt_map_result *result,
+		     uint8_t array_size);
+};
+
+struct map_common_param {
+	uint8_t id;
+	uint8_t len;
+	uint8_t data[0];
+};
+
+struct map_get_messages_listing_param {
+	uint16_t max_list_count;
+	uint8_t filter_message_type;
+	uint8_t reserved;
+	uint32_t parametr_mask;
+};
+
+struct bt_map_client {
+	struct bt_conn *conn;
+	char *path;
+	void *goep_client;
+	uint8_t state;
+	int8_t folder_level;
+	uint16_t max_list_cnt;
+	uint16_t get_list_cnt;
+	uint32_t mask;
+	uint8_t goep_connect: 1;
+	uint8_t path_valid: 1;
+	struct bt_map_client_cb *cb;
+};
+
+/** @brief MAP Client conect
+ *
+ *  @param conn Connection object.
+ *  @param path map message path.
+ *  @param cb bt_map_client_cb.
+ *
+ *  @return pointer to struct bt_map_client in case of success or NULL in case
+ *  of error.
+ */
+struct bt_map_client *bt_map_client_connect(struct bt_conn *conn, char *path,
+					    struct bt_map_client_cb *cb);
+
+/** @brief MAP Client disconnect
+ *
+ *  @param client The map client instance.
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_map_client_disconnect(struct bt_map_client *client);
+
+/** @brief MAP Client get map message
+ *
+ *  @param client The map client instance.
+ *  @param path  map message path..
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_map_client_get_message(struct bt_map_client *client, char *path);
+
+/** @brief MAP Client set map path
+ *
+ *  @param client The map client instance.
+ *  @param path  map message path..
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_map_client_setpath(struct bt_map_client *client, char *path);
+
+/** @brief MAP Client set map folder
+ *
+ *  @param client The map client instance.
+ *  @param path  map message path.
+ *  @param flags  map message flags..
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_map_client_set_folder(struct bt_map_client *client, char *path, uint8_t flags);
+
+/** @brief MAP Client get map meaasage list
+ *
+ *  @param client The map client instance.
+ *  @param max_list_count  map message max message count.
+ *  @param mask  map message mask..
+ *
+ *  @return 0 in case of success and error code in case of error.
+ */
+int bt_map_client_get_messages_listing(struct bt_map_client *client, uint16_t max_list_count,

```

Comment:
A lot of configurable parameters are missing here, how to expand it later? Such as, srm, srmp, name of folder, and application parameters, etc. We don't have to implement them now, but we need to consider how to expand them later.